### PR TITLE
Update datastore with device memory

### PR DESCRIPTION
**Pull request by pylessard**
Keeps the datastore updated by reading the device memory in a round robin scheme

### CODE REVIEW

File: python/scrutiny/core/variable.py [hunk 1]
```diff
@@ -21,6 +20,11 @@
         MASK_MAP[i] = v
 
 
+class Endianness(Enum):
+    Little = 0
+    Big = 1
```
[the remark this answers is not in the document]
Comment (fredericmorin) [reply python/scrutiny/core/variable.py:25]: for clarity. 
I just spent way too much time trying to figure the actual performance implication of using one over the other. This is probably a case of early over-optimisation but dont have much to support this.

File: python/scrutiny/core/variable.py [hunk 2]
```diff
@@ -262,7 +270,7 @@ def __init__(self, size: int):
             self.str = self.str_map[size]
 
         def decode(self, data: Union[bytes, bytearray], endianness: Endianness) -> int:
-            endianness_char = '<' if endianness == 'little' else '>'
+            endianness_char = '<' if endianness == Endianness.Little else '>'
```
Comment (fredericmorin) [inline python/scrutiny/core/variable.py:273]:
```suggestion
            endianness_char = '<' if endianness is Endianness.Little else '>'
```

Comment (pylessard) [reply python/scrutiny/core/variable.py:273]: Why?

File: python/scrutiny/core/variable.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pylessard) [reply python/scrutiny/core/variable.py:25]: hmm, doesn't seem over-engineered to have an enum that uses int. That's the point of the enum to give a name to a value

[the remark this answers is not in the document]
Comment (pylessard) [reply python/scrutiny/core/variable.py:25]: reading the doc, I think auto() should be the way to go.

**Pull request merged**